test/unit/test_symbolic.py

```diff
@@ -276,6 +275,74 @@ def test_min_max_known(self):
     assert max(1, a) == max(a, 1) == a
     assert min(1, a) == min(a, 1) == 1
 
+class TestSymRender(unittest.TestCase):
+  def test_sym_render(self):
+    a = Variable("a", 1, 8)
+    assert sym_render(a) == "a"
+    assert sym_render(1) == "1"
+    assert sym_render(a+1) == "(1+a)"
+
+class TestSymbolicSymbolicOps(unittest.TestCase):
+  def test_node_div_node(self):
+    i = Variable("i", 1, 10)
+    idx0 = Variable("idx0", 0, i*3-1)
+    assert NumNode(0) // (Variable("i", 1, 10)*128) == 0
+    assert NumNode(127) // (Variable("i", 1, 10)*128) == 0
+    assert idx0 // (i*3) == 0
+
+  def test_node_mod_node(self):
+    i = Variable("i", 1, 10)
+    idx0 = Variable("idx0", 0, i*3-1)
+    assert NumNode(0) % (Variable("i", 1, 10)*128) == 0
+    assert NumNode(127) % (Variable("i", 1, 10)*128) == 127
+    assert idx0 % (i*3) == idx0
+    assert i % i == 0
```

Comment:
can you add a test here that justifies `self - b == 0` over `self == b`?

I could think of only one "symbolic simplification" you mentioned in the main pr: `x*y - y*x == 0`. But looks like even `Node.sum()`/`SumNode` cannot simplify that.

I remember some `ProdNode` idea from earlier which may solve this by "sorting" just like `SumNode` does.

Reply:
oops, that would call `MulNode.__mod__()` not `Node.__mod__()`. But still, simplification is not done.

Reply:
Readability is the main concern here. *If* you *have* to do this "hack" to get more optimization can it not be centralized in `__eq__()` or `__lt__()` to keep the rest of the code clean?